presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -902,6 +911,15 @@ private PlanNode distinct(PlanNode node)
                 Optional.empty());
     }
 
+    private PlanNode addFilterNodeForAcidTable(TableScanNode node, Symbol isValidSymbol)
+    {
+        ComparisonExpression filterExpression = new ComparisonExpression(ComparisonExpression.Operator.EQUAL, isValidSymbol.toSymbolReference(), new BooleanLiteral("true"));
+        return new FilterNode(
```

Comment:
Do you need to expose `$isValid` to the engine?

Could you wrap ORC blocks in `LazyBlocks` that would return valid positions when the block is materialized?
This way you could avoid engine modifications and additional `FilterNode`

Reply:
I had thought of it earlier but picked FilterNode as the simpler solution which will be easy to extend to Parquet/other-formats too later if needed. I will try with the Block wrapper approach.

Reply:
Can we implement it as an extension of row filtering ? The connector will ask query engine to add a filter on `$isValid` using `TupleDomain`. But we need to check on how do we handle if filters are not applied on those tables ?

Reply:
> I had thought of it earlier but picked FilterNode as the simpler solution which will be easy to extend to Parquet/other-formats too later if needed.

I don't think there will be difference between file formats

Reply:
1. New Block: AcidBlock
	- This block has:
		i. Delegate block: The actual block read from disk. This block has rows both deleted and non-deleted.
		ii. Valid positions: List of positions of delegate block that are not deleted. This is created by cross-referencing rowIds of delegate block with delete_delta rowIds
	- This block offsets each request on a particular position to correct positions in delegate block
		- E.g . if position=0 is deleted then getByte(0) will be offsetted to getByte(1) on delegate block
		- This is done for all getters of different types in block: getByte, getShort, etc

2. For ACID tables, LazyLoader loads AcidBlock on load()

This is what i am planning to implement. Earlier, I was thinking of pushing valid position all the way into record reader in which case we wont need the new type AcidBlock but that 
would mean tight coupling with ORC. But with AcidBlock implementation, we can keep it format agnostic.

Reply:
Sorry, I meant `DictionaryBlock`, not `MapBlock`

Reply:
Updated with changes using DictionaryBlock